core/controllers/profile.py

```diff
@@ -664,22 +795,36 @@ def get(self):
             })
 
     @acl_decorators.open_access
-    def put(self):
+    def put(self) -> None:
         """Handles PUT requests."""
-        user_has_viewed_lesson_info_modal_once = self.normalized_payload.get(
-            'user_has_viewed_lesson_info_modal_once')
+        assert self.normalized_payload is not None
+        user_has_viewed_lesson_info_modal_once = self.normalized_payload[
+            'user_has_viewed_lesson_info_modal_once']
         if user_has_viewed_lesson_info_modal_once:
+            # Here we use MyPy ignore because .. confirm this part before
+            # merging the PR.
             user_services.set_user_has_viewed_lesson_info_modal_once(
-                self.user_id)
```

Comment:
ask.

Reply:
@vojtechjelinek @seanlip PTAL!

Reply:
This handler's PUT method is called here https://github.com/oppia/oppia/blob/f8a8d680c6f95dfc3c63a003484177ac06e4ce21/core/templates/domain/exploration/editable-exploration-backend-api.service.ts#L192

And recordLearnerHasViewedLessonInfoModalOnce is called when the user is logged in:
https://github.com/oppia/oppia/blob/f8a8d680c6f95dfc3c63a003484177ac06e4ce21/core/templates/pages/exploration-player-page/layout-directives/exploration-footer.component.ts#L380

Reply:
I would say, keep it open access but just do nothing in the case of self.user_id being None (and explain that in the comments).

Have you tried checking with @nithinrdy @paprajapati9 too?

Reply:
So this controller is only called for logged in user so I agree to what sean has suggested to keep it open access and do nothing if user id is None and explain it in a comment saying that this controller is only called when user is logged in so user id cannot be none. Thanks!

Reply:
Changes are added with comment PTAL!